### DISCUSSION TITLE
Automate setting the front matter "status" property of new/updated files?

### DISCUSSION THREAD

**Discussion by rmscode** [squidfunk/mkdocs-material#7198] [Q&A]
I recently learned of both "front matter" and it's `status` property. Before I use my very limited and "hacky" Javascript/Python skills to make something that is less than ideal, I was wondering if there is an existing method of automatically adding the front matter `status` property to new markdown files.

I also currently manually maintain a "Recent Additions and Revisions" section on my docs home page. It's just a bulleted list of recent stuff from the past 2ish weeks. Again, before I attempt to make something very hacky that can maybe generate that list for me, does something like this already exist? I was thinking of taking a look at [mkdocs-git-revision-date-localized-plugin](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin) by @timvink for ideas.

**Comment by kamilkrzyskow** [comment]
Hello @rmscode,
the status icon is rendered in the Jinja2 templates:
https://github.com/squidfunk/mkdocs-material/blob/7368c7f759e6163f7cce9af15d0b0d7bfc7a1d52/src/templates/partials/nav-item.html#L23-L39

So the `status` for a page needs to be defined before the template rendering is in progress, so `on_page_context` at the latest.
https://www.mkdocs.org/dev-guide/plugins/#on_page_context

You could use the file date from the git-revision plugin and add your own logic to it.
It uses the `on_page_markdown` event to add the date: 
https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/blob/b0ab74f9c4cba65f25cc3a28cd5f3234f297b5c0/mkdocs_git_revision_date_localized_plugin/plugin.py#L278

So if you know a bit of Python you could definitely make it work using a custom hook:
https://www.mkdocs.org/user-guide/configuration/#hooks
